### PR TITLE
jquery upgrade and navbox tweak

### PR DESCRIPTION
**Pull request by xithan**
select2 hack to make the new navbox look like an input field

### CODE REVIEW

File: card/mod/machines/lib/javascript/decko.js.coffee [hunk 1]
```diff
@@ -2,7 +2,7 @@ $.extend decko,
   # returns absolute path (starting with a slash)
   # if rawPath is relative (no slash), this adds relative root
   path: (rawPath) ->
-    if rawPath.match /^\//
+    if rawPath.match /^\/|:\/\//
```
Comment (xithan) [inline card/mod/machines/lib/javascript/decko.js.coffee]: with the updated jquery_ujs paths that start with "/" are completed to the full path. This method returned invalid paths like `/http://localhost:3000/...`

File: card/mod/pointer/lib/javascript/script_pointer_list_editor.js.coffee [hunk 2]
```diff
@@ -0,0 +1,48 @@
+$(window).ready ->
+  # add pointer item when clicking on "add another" button
+  $('body').on 'click', '.pointer-item-add', (event)->
+    decko.addPointerItem this
+    event.preventDefault() # Prevent link from following its href
```
Comment (ethn) [inline card/mod/pointer/lib/javascript/script_pointer_list_editor.js.coffee]: is this what you said is broken now?

File: card/mod/machines/lib/javascript/decko_filter.js.coffee [hunk 3]
```diff
@@ -66,7 +66,11 @@ updateSelectedCount = (el) ->
   count = selectedBin(el).children().length
   filterBox(el).find("._selected-items").html count
   deselectAllLink(el).attr "disabled", count == 0
-  addSelectedButton(el).attr "disabled", count == 0
+  if count > 0
+    addSelectedButton(el).removeClass("disabled")
+  else
+    addSelectedButton(el).addClass("disabled")
+
```
Comment (ethn) [inline card/mod/machines/lib/javascript/decko_filter.js.coffee]: is it correct that the button needed new handling but the deselectLink didn't?

File: card/mod/pointer/lib/javascript/script_pointer_list_editor.js.coffee [hunk 4]
```diff
@@ -0,0 +1,48 @@
+$(window).ready ->
```
Comment (ethn) [inline card/mod/pointer/lib/javascript/script_pointer_list_editor.js.coffee]: very cool that this is separated into the pointer mod.  woot.

File: card/mod/machines/lib/javascript/decko_filter.js.coffee (hunk 3 above)
Comment (xithan) [reply card/mod/machines/lib/javascript/decko_filter.js.coffee]: I don't know why an href attribute on a button worked before but it doesn't anymore so I changed the button to a link.

File: card/mod/pointer/lib/javascript/script_pointer_list_editor.js.coffee (hunk 2 above)
Comment (ethn) [reply card/mod/pointer/lib/javascript/script_pointer_list_editor.js.coffee]: oh, in a ticket that is already finished: https://www.pivotaltracker.com/story/show/157551476

Comment (xithan) [reply card/mod/pointer/lib/javascript/script_pointer_list_editor.js.coffee]: the problem was that the call for disabling the button was on window ready and not slot ready

**Pull request merged**